docs/features/incremental-generators.md

```diff
@@ -0,0 +1,485 @@
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+});
+```
+
+Note that `transformed` is similarly opaque. It represents the outcome of the
+transformation being applied to the data, but cannot be accessed directly.
+
+The transformed steps can be further transformed, and it is also valid to
+perform multiple transformations on the same input node, essentially 'splitting'
+the pipeline into multiple streams of processing.
+
+```csharp
+    // apply a 1-to-1 transform on each text, extracting the path
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+    // split the processing into two streams of derived data
+    var prefixTransform = transformed.Transform(path => "prefix_" + path);
+    var postfixTransform = transformed.Transform(path => path + "_postfixed");
+```
+
+### Batching
+
+In addition to the 1-to-1 transform shown above, there are also extension
+methods for producing and consuming batches of data. For instance a given
+transform may want to produce more than one value for each input, or want to
+view all the data in a single collected view in order to make cross data
+decisions.
+
+``` csharp
+public static partial class IncrementalValueSourceExtensions
+{
+    // 1 => many (or none)
+    public static IncrementalValueSource<U> TransformMany<T, U>(this IncrementalValueSource<T> source, Func<T, IEnumerable<U>> func) => ...
+
+    // many => 1
+    public static IncrementalValueSource<U> BatchTransform<T, U>(this IncrementalValueSource<T> source, Func<IEnumerable<T>, U> func) => ...
+
+    // many => many (or none)
+    public static IncrementalValueSource<U> BatchTransformMany<T, U>(this IncrementalValueSource<T> source, Func<IEnumerable<T>, IEnumerable<U>> func) => ...
+}
+```
+
+In our above example we could use `BatchTransform` to collect the individual
+file paths collected, and convert them into a single collection:
+
+```csharp
+    // apply a 1-to-1 transform on each text, which represents extracting the path
+    var transformed = additionalTexts.Transform((text) => text.Path);
+
+    // batch the collected file paths into a single collection
+    var batched = transformed.BatchTransform(texts => texts);
+```
+
+The author could have equally combined these two steps into a single operation
+that utilizes LINQ:
+
+```csharp
+    // using System.Linq;
+    var singleOp = additionalTexts.BatchTransform(texts => texts.Select(text => text.Path));
+```
+
+**OPEN QUESTION** Should there be versions of
+`BatchTransform`/`BatchTransformMany` that take no transformation, and just
+perform the identity function as specified above?
+
+### Outputting values
+
+At some point in the pipeline the author will want to actually use the
+transformed data to produce an output, such as a `SourceText`. For this purpose
+there are 'terminating' extension methods that allow the author to provide the
+resulting data the generator produces. For instance there is a
+`GenerateSource(...)` method:
```

Comment:
do you list all the GneerateXXX methods we supply?  If so, if it's a reasonable list, i recommend front loading the list of their names here to make it clear what sort of things can be generated.